src/osd/PG.cc

```diff
@@ -5531,6 +5532,7 @@ void PG::handle_advance_map(
   recovery_state.handle_event(evt, rctx);
   if (pool.info.last_change == osdmap_ref->get_epoch())
     on_pool_change();
+  update_store_with_options();
```

Comment:
Ah, it looks like advance_map will always happen (eventually) when the OSD is marked up, but I think it would be better to put this call inside on_pool_change() (so that it only happens when pool options change, not on every epoch change) *and* in the boost::statechart::result PG::RecoveryState::Initial::react(const Load& l) method (which happens explicitly on PG load).